client/src/connection/itc/script.ts

```diff
@@ -106,6 +124,29 @@ class SASRunner{
       Write-Host $log
     } while ($log.Length -gt 0)
   }
-}
 
+  [void]FetchResultsFile([string]$filePath, [string]$outputFile) {
+    $fileRef = ""
+    $objFile = $this.objSAS.FileService.AssignFileref("outfile", "DISK", $filePath, "", [ref] $fileRef)
+    $objStream = $objFile.OpenBinaryStream(1);
+    [Byte[]] $bytes = 0x0
+
+    $endOfFile = $false
+    $byteCount = 0
+    $outStream = [System.IO.StreamWriter] $outputFile
+    do
+    {
+      $objStream.Read(1024, [ref] $bytes)
+      $outStream.Write([System.Text.Encoding]::UTF8.GetString($bytes))
```

Comment:
This is a problem. A character in UTF8 may occupy various bytes (from 1 to 4 bytes). The 1024th byte may not be an end of a character. So when you try to get string from the 1024 bytes, the last character may be broken. And when it happens, the first character get from the next 1024 bytes will also be broken.
I believe it is the cause of the issue Zhirong is seeing. It only happens at cases where 1024 happened to cut a character.
It is a regression compared to v1.5 as it also affects local connection. In v1.5 it directly read local file thus no such issue. So I think it would be better to fix it in this PR.
We should collect all bytes before converting it to string.

Reply:
Resolved in https://github.com/sassoftware/vscode-sas-extension/pull/592/commits/ad48cd15c82de5327db9f50411a6e69992178f93